internals/reminders.py

```diff
@@ -0,0 +1,179 @@
+        current_milestone_info)
+    notifier.send_emails(email_tasks)
+    return {'message': f'{len(email_tasks)} email(s) sent or logged.'}
+
+  def prefilter_features(self, current_milestone_info, features):
+    """Return a list of features that fit class-specific criteria."""
+    return features  # Defaults to no prefiltering.
+
+  def filter_by_milestones(self, current_milestone_info, features):
+    """Return [(feature, milestone)] for features with a milestone in range."""
+    # 'current' milestone is the next stable milestone that hasn't landed.
+    # We send notifications to any feature planned for beta or stable launch
+    # in the next N weeks. E.g., beta for (current + 2) starts in roughly 8
+    # weeks.  So we check if any features have launches in these 3 milestones
+    # (current, current + 1, and current + 2).
+    min_mstone = int(current_milestone_info['mstone'])
+    max_mstone = min_mstone + self.FUTURE_MILESTONES_TO_CONSIDER
+
+    result = []
+    for feature in features:
+      field_values = [getattr(feature, field) for field in self.MILESTONE_FIELDS]
+      matching_values = [
+          m for m in field_values
+          if m is not None and m >= min_mstone and m <= max_mstone]
+      if matching_values:
+        result.append((feature, min(matching_values)))
+
+    return result
+
+  def determine_features_to_notify(self, current_milestone_info):
+    """Get all features filter them by class-specific and milestone criteria."""
+    features = core_models.Feature.query(
+        core_models.Feature.deleted == False).fetch(None)
+    prefiltered_features = self.prefilter_features(
+        current_milestone_info, features)
+    features_milestone_pairs = self.filter_by_milestones(
+        current_milestone_info, prefiltered_features)
+    return features_milestone_pairs
+
+
+class FeatureAccuracyHandler(AbstractReminderHandler):
+  """Periodically remind owners to verify the accuracy of their entries."""
+
+  ACCURACY_GRACE_PERIOD = timedelta(weeks=4)
+  SUBJECT_FORMAT = '[Action requested] Update %s'
+  EMAIL_TEMPLATE_PATH = 'accuracy_notice_email.html'
+  FUTURE_MILESTONES_TO_CONSIDER = 2
+  MILESTONE_FIELDS = (
+      'dt_milestone_android_start',
+      'dt_milestone_desktop_start',
+      'dt_milestone_ios_start',
+      'dt_milestone_webview_start',
+      'ot_milestone_android_start',
+      'ot_milestone_desktop_start',
+      'ot_milestone_webview_start',
+      'shipped_android_milestone',
+      'shipped_ios_milestone',
+      'shipped_milestone',
+      'shipped_webview_milestone')
+
+  def prefilter_features(self, current_milestone_info, features):
+    now = datetime.now()
+    prefiltered_features = [
+        feature for feature in features
+        # It needs review if never reviewed, or if grace period has passed.
+        if (feature.accurate_as_of is None or
+            feature.accurate_as_of + self.ACCURACY_GRACE_PERIOD < now)]
```

Comment:
Note that I reversed the sense of this condition (now it evaluates to True for features that should notify whereas you had it it True to skip a a feature), but I kept the comparison as less-than. When looking at this code for this PR, I thought that the original less than was actually a defect. Please double check that I have this right now.

Reply:
Updated